### PR TITLE
adding entity framework sample

### PR DESCRIPTION
**Pull request by jeffhollan**

### CODE REVIEW

File: samples/EntityFramework/.gitignore [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+*.sln.iml
+
+# CodeRush
+.cr/
+
+# Python Tools for Visual Studio (PTVS)
+__pycache__/
+*.pyc
+
+Migrations/
```
Comment (fabiocav) [inline samples/EntityFramework/.gitignore:266]: Assuming this is the main reason for the file, correct? If so, let's just update the root .gitignore instead to make sure this is there.

File: samples/EntityFramework/HttpTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+using System.IO;
```
Comment (fabiocav) [inline samples/EntityFramework/HttpTrigger.cs:1]: nit: header:
```
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

```

File: samples/EntityFramework/HttpTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Hollan.Function
```
Comment (fabiocav) [inline samples/EntityFramework/HttpTrigger.cs:12]: Fix namespaces

File: samples/EntityFramework/HttpTrigger.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        public HttpResponseData GetPosts(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "posts")] HttpRequestData req,
+            FunctionContext context)
+        {
+           var logger = context.GetLogger("GetPosts");
+           logger.LogInformation("C# HTTP GET/posts trigger function processed a request.");
+
+           var postsArray = _context.Posts.OrderBy(p => p.Title).ToArray();
+           var response = req.CreateResponse(HttpStatusCode.OK);
+           response.WriteString(JsonConvert.SerializeObject(postsArray));
```
Comment (fabiocav) [inline samples/EntityFramework/HttpTrigger.cs:32]: Use the new JSON extension APIs (`WriteAsJsonAsync`)

File: samples/EntityFramework/Models.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Design;
+
+namespace Hollan.Function
```
Comment (fabiocav) [inline samples/EntityFramework/Models.cs:6]: Comments about namespace/header applies to other files.

File: samples/EntityFramework/HttpTrigger.cs (hunk 4 above)
Comment (jeffhollan) [reply samples/EntityFramework/HttpTrigger.cs:32]: @fabiocav has that been published yet? It's not showing for me in the `preview5`

Comment (fabiocav) [reply samples/EntityFramework/HttpTrigger.cs]: Ah, yeah, I think the samples were updated to use a package reference here.

Will look at the updates, but I see you had a follow up. Did you update the reference to get the new APIs?

Comment (jeffhollan) [reply samples/EntityFramework/HttpTrigger.cs]: OH can do - do we prefer we use a package reference to the source or keep it referencing NuGet?  I'm ok either way, only benefit of the latter is you can "copy paste" out of the repo and it all still works without updating references, but I can reference to base.  LMK what you prefer

File: samples/EntityFramework/net-5-entity-framework.csproj [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    <LangVersion>preview</LangVersion>
+    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
+    <OutputType>Exe</OutputType>
+    <_FunctionsSkipCleanOutput>true</_FunctionsSkipCleanOutput>
+    <RootNamespace>net_5_entity_framework</RootNamespace>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.12-preview1" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.0.1-preview5" OutputItemType="Analyzer" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.0.0-preview5" />
```
Comment (mhoeger) [inline samples/EntityFramework/net-5-entity-framework.csproj:13]: Bump out of preview 🎉

File: samples/EntityFramework/HttpTrigger.cs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            logger.LogInformation("C# HTTP POST/blog trigger function processed a request.");
+               
+            var post = JsonConvert.DeserializeObject<Post>(await new StreamReader(req.Body).ReadToEndAsync());
+            post.BlogId = id;
+            
+            var entity = await _context.Posts.AddAsync(post);
+            await _context.SaveChangesAsync(CancellationToken.None);
+
+            var response = req.CreateResponse(HttpStatusCode.OK);
+            response.WriteString(JsonConvert.SerializeObject(entity.Entity));
```
Comment (mhoeger) [inline samples/EntityFramework/HttpTrigger.cs:73]: nit: `WriteAsJsonAsync(entity.Entity)`

File: samples/EntityFramework/HttpTrigger.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        [Function("CreatePost")]
+        public async Task<HttpResponseData> CreatePostAsync(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "blog/{id}/post")] HttpRequestData req,
+            int id,
+            FunctionContext context)
+        {
+            var logger = context.GetLogger("CreatePost");
+            logger.LogInformation("C# HTTP POST/blog trigger function processed a request.");
+               
+            var post = JsonConvert.DeserializeObject<Post>(await new StreamReader(req.Body).ReadToEndAsync());
```
Comment (mhoeger) [inline samples/EntityFramework/HttpTrigger.cs:66]: nit: `req.ReadFromJsonAsync <Post>()`

File: samples/EntityFramework/HttpTrigger.cs [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        public HttpResponseData GetPosts(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "posts")] HttpRequestData req,
+            FunctionContext context)
+        {
+           var logger = context.GetLogger("GetPosts");
+           logger.LogInformation("C# HTTP GET/posts trigger function processed a request.");
+
+           var postsArray = _context.Posts.OrderBy(p => p.Title).ToArray();
+           var response = req.CreateResponse(HttpStatusCode.OK);
+           response.WriteAsJsonAsync()
```
Comment (mhoeger) [inline samples/EntityFramework/HttpTrigger.cs:34]:
```suggestion
           response.WriteAsJsonAsync(postsArray);
```

File: samples/EntityFramework/Program.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+// #endif
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration(c =>
+                {
+                    c.AddCommandLine(args);
+                })
+                .ConfigureFunctionsWorker((c, b) =>
+                {
+                    b.UseFunctionExecutionMiddleware();
+                })
```
Comment (mhoeger) [inline samples/EntityFramework/Program.cs:31]:
```suggestion
                .ConfigureFunctionsWorkerDefaults()
```

**Review by mhoeger — COMMENTED** [review]
Suggestions to align with API changes that happened between preview and GA

**Comment by fabiocav** [comment]
@jeffhollan sorry about moving things around as you were writing this. Now that things are released and will be stable, want to update this so we can get the sample in?

**Comment by jeffhollan** [comment]
@fabiocav @mhoeger ok - made all suggested changes. I think this is ready to go now. LMK

**Review by mhoeger — APPROVED** [review]
lgtm!

**Review by fabiocav — APPROVED** [review]

**Pull request merged**